### ISSUE TITLE
Vault stooped working for MasterLess in version 2017.7.3 

### ISSUE THREAD

**Issue by 1ricardo** [saltstack/salt#46138]
### Description
After upgrade salt-minion to version 2017.7.3, there is problem with generation the token for Vault.
What is important SALT is working in **MasterLess** mode, after downgrading to version 2017.7.2 everything working normally.

#### version 2017.7.2
```sh
$ salt-call vault.read_secret 'secret/ssh/user1'
 local:
   ----------  
   password:  
       ”abc123″
```

#### version 2017.7.3
```sh 
$ salt-call vault.read_secret 'secret/ssh/user1'
[ERROR   ] Failed to get token from master! Response is not a dict: No access to master. If using salt-call with --local, please remove.
[ERROR   ] Failed to read secret! CommandExecutionError: No access to master. If using salt-call with --local, please remove.
Error running 'vault.read_secret': No access to master. If using salt-call with --local, please remove.
```

### Setup
```sh
vault:
  url: http://vault.service.consul:8200
  auth:
    method: token
    token: 11111111-2222-3333-4444-555555555555
  policies:
    - saltstack

file_client: local

file_roots:
  base:
    - /path/to/local/state

pillar_roots:
  base:
    - /path/to/local/pillar
```
As an additional configuration, the Consul service is also configured and it's working properly.

### Versions Report
```sh
~ # salt-call --versions-report
Salt Version:
           Salt: 2017.7.3
 
Dependency Versions:
           cffi: 1.11.4
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.3.1
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 3.2.5
 
System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.11.6.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

**Comment by Ch3LL** [comment]
can you see if this fixes it for you ? https://github.com/saltstack/salt/pull/45981

**Comment by 1ricardo** [comment]
Hi @Ch3LL,

Thank You for respond. I changed the configuration according with your suggestion, but unfortunately I have the same error message on version **2017.7.3**
#### Config
```sh
...
file_client: local
local: True
...
```

#### Respond
```sh
$ salt-call vault.read_secret 'secret/ssh/user1'
[ERROR   ] Failed to get token from master! Response is not a dict: No access to master. If using salt-call with --local, please remove.
[ERROR   ] Failed to read secret! CommandExecutionError: No access to master. If using salt-call with --local, please remove.
Error running 'vault.read_secret': No access to master. If using salt-call with --local, please remove.
```

**Comment by Ch3LL** [comment]
did you also try the patch here: https://github.com/saltstack/salt/pull/45981/files#diff-e75b02a233b726e8be8f614a7a554fd5 ?

**Comment by 1ricardo** [comment]
No I didn't, so later I patched **salt/config/__init__.py** without any problems, but I had problem with **salt/utils/vault.py**. I noticed that, there is more differences in context, between file from link [https://github.com/gtmanfred/salt/blob/0d448457dccc67e92627f35dfa7e2de963a4d3d1/salt/utils/vault.py](https://github.com/gtmanfred/salt/blob/0d448457dccc67e92627f35dfa7e2de963a4d3d1/salt/utils/vault.py) and file from disk.

#### Here is diff
```sh
--- old_vault.py	2018-02-22 20:39:28.140076012 +0100
+++ new_vault.py	2018-02-22 20:59:25.432836368 +0100
@@ -98,7 +98,7 @@
     Get the connection details for calling Vault, from local configuration if
     it exists, or from the master otherwise
     '''
-    if 'vault' in __opts__ and __opts__.get('__role', 'minion') == 'master':
+    def _use_local_config():
         log.debug('Using Vault connection details from local config')
         try:
             return {
@@ -108,6 +108,11 @@
         except KeyError as err:
             errmsg = 'Minion has "vault" config section, but could not find key "{0}" within'.format(err.message)
             raise salt.exceptions.CommandExecutionError(errmsg)
+
+    if 'vault' in __opts__ and __opts__.get('__role', 'minion') == 'master':
+        return _use_local_config()
+    elif any((__opts__['local'], __opts__['file_client'] == 'local', __opts__['master_type'] == 'disable')):
+        return _use_local_config()
     else:
         log.debug('Contacting master for Vault connection details')
         return _get_token_and_url_from_master()
```
however after coping whole context from link to **salt/utils/vault.py**, the command  **salt-call vault** worked again in **MasterLess** mode

```sh
# salt-call vault.read_secret 'secret/ssh/user1'
local:
    ----------
    password:
        ”abc123″
```

I am using following repo [https://repo.saltstack.com/yum/redhat/salt-repo-latest-2.el7.noarch.rpm ](https://repo.saltstack.com/yum/redhat/salt-repo-latest-2.el7.noarch.rpm) on Centos 7

**Comment by Ch3LL** [comment]
K perfect. This patch will be included in the upcoming 2017.7.4 release. We hope to release it really soon, possibly next week as long as testing is okay. I will close this now as it looks like this PR does revolve the issue for you thanks :)

**Comment by 1ricardo** [comment]
Thank You @Ch3LL for your support and help.
